### PR TITLE
CoverArtUtils: Fix Reload From File/Folder, Updates Wrong Cover Art

### PR DESCRIPTION
**Pull request by fatihemreyildiz**
This PR aims to solve this bug mentioned on [Launchpad](https://bugs.launchpad.net/mixxx/+bug/1985843).

This simply fixes to wrong cover art update if the names has more than one "." dot. I
track.1.mp3
track.2.mp3
track.3.mp3
track.4.jpg
track.5.jpg
 
If track.1-track.2-track.3's cover arts updated via "Reload from file/folder" it updates it with track.4.jpg. 

With this change, it looks for the complete full name, so track.1.mp3 won't be updated with track.4.jpg. 

But with that change, this can change the old behavior, if the files named as;
track.1.mp3.jpg
track.1.mp3
updating wouldn't not update the cover art, to not change the behavior of old updating, there is an another block added. If complete full name is not working, this will look for file name of the track `"track.1.mp3"` and complete name of the cover art `"track.1.mp3".jpg`

### CODE REVIEW

**Comment by daschuer** [comment]
Thank you. This looks like a bug fix for 2.3.

**Comment by fatihemreyildiz** [comment]
Yes, that is happening in 2.3 :+1:

File: src/library/coverartutils.cpp [hunk 1]
```diff
@@ -116,13 +116,17 @@ CoverInfoRelative CoverArtUtils::selectCoverArtForTrack(
...
-                    coverBaseName.compare(trackFile.baseName(),
+                    coverBaseName.compare(trackFile.completeBaseName(),
                             Qt::CaseInsensitive) == 0) {
                 bestInfo = &file;
                 // This is the best type (TRACK_BASENAME) so we know we're done.
                 break;
+            } else if (bestType > TRACK_BASENAME &&
+                    coverBaseName.compare(trackFile.fileName(),
+                            Qt::CaseInsensitive) == 0) {
+                bestInfo = &file;
```
Comment (daschuer) [inline src/library/coverartutils.cpp:129]:
```suggestion
                bestType = TRACK_BASENAME;
                bestInfo = &file;
```

**Review by daschuer — CHANGES_REQUESTED** [review]
One missing line.

Can you rebase this on 2.3 than. Thank you.

File: src/library/coverartutils.cpp (hunk 1 above)
Comment (fatihemreyildiz) [reply src/library/coverartutils.cpp:121]: I've added the missing line.

**Review by daschuer — APPROVED** [review]
LGTM

**Comment by daschuer** [comment]
Does anyone else want to have a look?

**Comment by daschuer** [comment]
Thank you!

**Comment by daschuer** [comment]
Ups, this is now broken in the 2.3 branch: https://github.com/mixxxdj/mixxx/runs/7979410003
I did not realize that the CI was still testing the main merge after changing the merge target. 
I will push a fix directly to 2.3

**Comment by daschuer** [comment]
https://github.com/mixxxdj/mixxx/commit/517d7a64a51d439be309f6fcbcd73cce4820075e

**Comment by fatihemreyildiz** [comment]
Thank you for taking care. 

I guess the file `trackfile.h` in 2.3 branch, changed to `fileinfo.h` in main branch right? I was trying to fix it and added the completeBaseName in `file.h` in main branch and while re-basing to 2.3 there was no `fileinfo.h` so I deleted completeBaseName, and compiled successfully. 

I just didn't get it why it failed. I hope I didn't do something missing-wrong.

**Pull request merged**